src/library/components/resource-lightbox.js

```diff
@@ -59,6 +57,19 @@ var ResourceLightbox = Component({
   },
 
   replaceResource: function (resource) {
+    let params = ParseQueryString()
+    let openAssign = params.openAssign
+    let redirect = params.redirecting_after_sign_in
+    let savedUrl = window.location.toString().replace(/\?.*/, '')
+
+    if (redirect) {
+      // Currently, a redirect after login will always return you to the
+      // lightbox over the home page even if you logged in at a collection page.
+      // If that gets fixed, this should be updated to account for the
+      // possibility the lightbox is being shown on a collection page.
+      savedUrl = '/'
+    }
+
```

Comment:
I'm not sure this is going to work very well. I looked into why the url is the root in this case:
- user loads url that opens a lightbox directly like: https://learn.concord.org/resources/799/3d-exploration-of-bound-antibody-and-antigen
- then the user closes the lightbox
- now the browser url is the root

This happens because of this line: https://github.com/concord-consortium/rigse/blob/a7eb234951ef2ea0d9add4d6f575ce001be41e9b/app/views/home/home.html.haml#L29

With the switching to state that property setting in the portal code will have no effect. 

The redirect check above would also not pick up a direct load of a resource. So with this new code above I'd expect:
- user loads url that opens a lightbox directly like: https://learn.concord.org/resources/799/3d-exploration-of-bound-antibody-and-antigen
- then the user closes the lightbox
- now the browser url is still the lightbox url (no change)

 A quick thought is to add a `savedUrl` parameter to the URL. If the `savedUrl` parameter isn't set then use `/`.  So now on a collection page the savedUrl parameter would be set to the collection page.  If the user chooses to login then they will be redirected back to the resource lightbox with this same savedUrl parameter so when they close the page they will still be on the collection page. The downside of this is that the URL for a resource lightbox on a collection page will be different than the same lightbox on the stem resource finder. So if someone shares it they  have this extra param in it. That might be OK or a feature, or it could be problematic if you want all sharing to be to a single place.  If you do implement this I'd suggest changing the name from savedUrl to background or something that is more clear that this is the url underneath.  It should also forced to be a just a path, not a full URL. It is insecure to let people construct URLs that redirect to other sites.

Reply:
@scytacki I've made some changes, but I still feel like things are still a little kludgy and getting more complicated in that regard, too. There are some accompanying portal code changes at https://github.com/concord-consortium/rigse/pull/724

For what it's worth, I did test out the previous work and it all seemed to function properly, including in the second situation you described. From what I could tell, it didn't seem to introduce any new bugs concerning the URLs.

The latest changes fix at least some of those URL bugs though, again, I feel like the fixes are iffy. I suspect it's the case that I should stop trying to make this work with things as they are  and start moving everything to portal-pages.

Reply:
It seems like we should work on this together.  I don't want to make it more complicated. If it was working with your code from before then I'm confused so I'd like to understand it.